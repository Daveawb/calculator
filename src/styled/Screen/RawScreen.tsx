import * as React from 'react';

type RawScreenProps = {
    displayValue: string;
    className?: string;
};

export const RawScreen: React.FC<RawScreenProps> = (props: RawScreenProps) => {
    const { className, displayValue } = props;

    return (
        <div className={className}>
            <span>{displayValue}</span>
        </div>
    );
};
