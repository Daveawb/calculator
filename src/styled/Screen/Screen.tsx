import styled from 'styled-components';
import { RawScreen } from './RawScreen';

export const Screen = styled(RawScreen)`
    flex-grow: 1;
    font-size: 4rem;
    text-align: left;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
    margin-right: 4px;
    margin-left: 4px;
    padding-left: 0.5rem;
    border: 2px solid #eee;
    background-color: #fff;
    border-radius: 1rem;
    overflow: hidden;
`;
