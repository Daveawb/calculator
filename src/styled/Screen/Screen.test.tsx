import { shallow } from 'enzyme';
import * as React from 'react';
import { Screen } from './Screen';

describe('<Screen />', () => {
    test('matches snapshot', () => {
        const wrapper = shallow(<Screen displayValue="test" />);
        expect(wrapper).toMatchSnapshot();
    });
});
