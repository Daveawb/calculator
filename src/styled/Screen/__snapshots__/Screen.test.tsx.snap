// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> matches snapshot 1`] = `
<StyledComponent
  displayValue="test"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    flex-grow: 1;
    font-size: 4rem;
    text-align: left;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
    margin-right: 4px;
    margin-left: 4px;
    padding-left: 0.5rem;
    border: 2px solid #eee;
    background-color: #fff;
    border-radius: 1rem;
    overflow: hidden;
",
        ],
      },
      "displayName": "Styled(RawScreen)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "self": [Circular],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
